src/model/src/PrismaticJoint.cpp

```diff
@@ -0,0 +1,358 @@
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ *
+ */
+
+#include <iDynTree/Model/PrismaticJoint.h>
+
+#include <iDynTree/Core/Axis.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/TransformDerivative.h>
+#include <iDynTree/Model/LinkState.h>
+#include <iDynTree/Core/Wrench.h>
+#include <iDynTree/Core/Twist.h>
+
+#include <cassert>
+
+#include <cfloat>
+
+namespace iDynTree
+{
+
+PrismaticJoint::PrismaticJoint(const LinkIndex _link1, const LinkIndex _link2,
+                             const Transform& _link1_X_link2, const Axis& _translation_axis_wrt_link1):
+                             link1(_link1), link2(_link2), link1_X_link2_at_rest(_link1_X_link2),
+                             translation_axis_wrt_link1(_translation_axis_wrt_link1)
+{
+    this->setPosCoordsOffset(0);
+    this->setDOFsOffset(0);
+
+    this->resetAxisBuffers();
+    this->resetBuffers(0);
+    this->disablePosLimits();
+}
+
+PrismaticJoint::PrismaticJoint(const PrismaticJoint& other):
+                             link1(other.link1), link2(other.link2),
+                             link1_X_link2_at_rest(other.link1_X_link2_at_rest),
+                             translation_axis_wrt_link1(other.translation_axis_wrt_link1),
+                             m_hasPosLimits(other.m_hasPosLimits),
+                             m_minPos(other.m_minPos), m_maxPos(other.m_maxPos)
+{
+    this->setPosCoordsOffset(other.getPosCoordsOffset());
+    this->setDOFsOffset(other.getDOFsOffset());
+
+    this->resetAxisBuffers();
+    this->resetBuffers(0);
+}
+
+PrismaticJoint::~PrismaticJoint()
+{
+}
+
+IJoint* PrismaticJoint::clone() const
+{
+    return (IJoint *) new PrismaticJoint(*this);
+}
+
+
+void PrismaticJoint::setRestTransform(const Transform& _link1_X_link2)
+{
+    this->link1_X_link2_at_rest = _link1_X_link2;
+
+    this->resetAxisBuffers();
+}
+
+LinkIndex PrismaticJoint::getFirstAttachedLink() const
+{
+    return link1;
+}
+
+LinkIndex PrismaticJoint::getSecondAttachedLink() const
+{
+    return link2;
+}
+
+Transform PrismaticJoint::getRestTransform(const LinkIndex p_linkA, const LinkIndex p_linkB) const
+{
+    if( p_linkA == link1 )
+    {
+        assert(p_linkB == link2);
+        return link1_X_link2_at_rest;
+    }
+    else
+    {
+        assert(p_linkA == link2);
+        assert(p_linkB == link1);
+        return link1_X_link2_at_rest.inverse();
+    }
+}
+
+void PrismaticJoint::updateBuffers(const double new_q) const
+{
+    if( new_q != q_previous )
+    {
+        resetBuffers(new_q);
+    }
+
+    return;
+}
+
+void PrismaticJoint::resetBuffers(const double new_q) const
+{
+    this->link1_X_link2 = translation_axis_wrt_link1.getTranslationTransform(new_q)*link1_X_link2_at_rest;
+    this->link2_X_link1 = link1_X_link2.inverse();
+
+    q_previous = new_q;
+}
+
+void PrismaticJoint::resetAxisBuffers() const
+{
+    this->S_link1_link2 = -(translation_axis_wrt_link1).getTranslationTwist(1.0);
```

Comment:
I tried to reason where there is a `-` , I dont think it is needed as the getTranslationTwist() method sets linVel by simply multiplying dir vector with the argument passed, see https://github.com/robotology/idyntree/pull/269/commits/531830c30efae6ed2f143e5d7aa66291a4463f9e. 
If this is good, you can close this PR. It has been a long time :smile: